### PR TITLE
New design for lack screens

### PR DESCRIPTION
**Pull request by KeepCalmAndProgram**

### CODE REVIEW

File: lib/presentation/widgets/icon_text_button_widget.dart [hunk 1]
```diff
@@ -0,0 +1,63 @@
+import 'package:flutter/material.dart';
+
+class IconTextButton extends StatefulWidget {
+  final String text_left_side;
```
Comment (alex-tifox) [inline lib/presentation/widgets/icon_text_button_widget.dart]: lowerCamelCase here

File: lib/presentation/widgets/text_button_widget.dart [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+class _TextButtonWidgetState extends State<TextButtonWidget> {
+  @override
+  Widget build(BuildContext context) {
+    return TextButton(
+      style: TextButton.styleFrom(
+        textStyle: Theme.of(context)
+            .textTheme
+            .bodyText1, //const TextStyle(fontSize: 20),
+      ),
+      onPressed: () {},
```
Comment (alex-tifox) [inline lib/presentation/widgets/text_button_widget.dart:19]: you should pass here VoidCallback, to expose onPressed for another places you want to use TextButtonWidget. 
Like:
`final VoidCallback onPressed;`
In TextButton:
`onPressed: onPressed`

File: lib/presentation/widgets/text_button_widget.dart [hunk 3]
```diff
@@ -0,0 +1,31 @@
+import 'package:flutter/material.dart';
+
+class TextButtonWidget extends StatefulWidget {
```
Comment (alex-tifox) [inline lib/presentation/widgets/text_button_widget.dart]: Why this is made as StatefulWidget

File: lib/presentation/widgets/icon_text_button_widget.dart [hunk 4]
```diff
@@ -0,0 +1,63 @@
+import 'package:flutter/material.dart';
+
+class IconTextButton extends StatefulWidget {
```
Comment (alex-tifox) [inline lib/presentation/widgets/icon_text_button_widget.dart]: Why this is made as StatefulWidget -> change to Stateless, there is no point of using stateful

File: lib/presentation/widgets/icon_text_button_widget.dart (hunk 1 above)
Comment (alex-tifox) [reply lib/presentation/widgets/icon_text_button_widget.dart]: and everywhere in the project when you are declaring variables, pay more attention on that

File: lib/presentation/screens/search_screen.dart [hunk 5]
```diff
@@ -0,0 +1,117 @@
...
+            decoration: BoxDecoration(
+              borderRadius: BorderRadius.circular(20.0),
+            ),
+            child: ListView.separated(
+              scrollDirection: Axis.horizontal,
+              padding: const EdgeInsets.all(12),
+              itemCount: 5,
+              separatorBuilder: (context, index) => const SizedBox(width: 12),
+              itemBuilder: (context, index) {
+                return PhotoView(width: _width, height: _height);
```
Comment (alex-tifox) [inline lib/presentation/screens/search_screen.dart]: you could use arrow function here for syntax sugar

File: lib/presentation/screens/search_screen.dart [hunk 6]
```diff
@@ -0,0 +1,117 @@
...
```
Comment (alex-tifox) [inline lib/presentation/screens/search_screen.dart]: arrow function here as syntax sugar

File: lib/presentation/screens/search_screen.dart [hunk 7]
```diff
@@ -0,0 +1,117 @@
...
+                  contentPadding: const EdgeInsets.symmetric(vertical: 15.0),
+                  fillColor: Colors.white,
+                  filled: true,
+                  border: OutlineInputBorder(
+                      borderRadius: BorderRadius.circular(20.0),
+                      borderSide: const BorderSide(width: 0.8)),
+                  enabledBorder: OutlineInputBorder(
+                      borderRadius: BorderRadius.circular(30.0),
+                      borderSide: BorderSide(
+                          width: 0.8, color: Theme.of(context).primaryColor)),
```
Comment (alex-tifox) [inline lib/presentation/screens/search_screen.dart]: forgotten comma

File: lib/presentation/screens/photo_view_screen.dart [hunk 8]
```diff
@@ -65,14 +65,9 @@ class _PhotoViewScreenState extends State<PhotoViewScreen> {
...
-                    child: const Text(
+                    child: Text(
                       '12 July 2022',
-                      style: TextStyle(
-                        // change to best version of text style in main !!!
-                        fontSize: 20,
-                        fontWeight: FontWeight.bold,
-                        color: Colors.white,
-                      ),
+                      style: Theme.of(context).textTheme.bodyText1,
```
Comment (alex-tifox) [inline lib/presentation/screens/photo_view_screen.dart]: good using of Theme

File: lib/presentation/screens/photo_gallery_screen.dart [hunk 9]
```diff
@@ -17,26 +17,41 @@ class _PhotoGalleryScreenState extends State<PhotoGalleryScreen> {
...
+              PhotoView(width: _width, height: _height),
+              PhotoView(width: _width, height: _height),
+              PhotoView(width: _width, height: _height),
+            ],
+          ),
+        ),
+      ),
+      floatingActionButton: FloatingActionButton(
+        onPressed: () {
+          //_showChoiceDialog(context);
```
Comment (alex-tifox) [inline lib/presentation/screens/photo_gallery_screen.dart]: don't see this choice dialog function in this file

File: lib/presentation/screens/albums_screen.dart [hunk 10]
```diff
@@ -19,201 +23,133 @@ class _AlbumsScreenState extends State<AlbumsScreen> {
...
+              ),
             ),
-          ), // for Recent widget
-        ],
+            Column(
+              children: [
+                IconTextButton(
+                    text_left_side: 'Video',
+                    text_right_side: '17 >',
+                    icon: Icons.camera_alt_outlined),
```
Comment (alex-tifox) [inline lib/presentation/screens/albums_screen.dart]: forgotten comma

File: lib/main.dart [hunk 11]
```diff
@@ -27,18 +29,16 @@ class MyApp extends StatelessWidget {
...
+        textTheme: const TextTheme(
           bodyText1: TextStyle(
+            fontSize: 20,
+            fontWeight: FontWeight.bold,
             color: Colors.black,
-            fontSize: 15,
-
-            /// find best size for text !
           ),
           bodyText2: TextStyle(
```
Comment (alex-tifox) [inline lib/main.dart:42]: why the TextTheme above is const, but this one isn't?

**Review by alex-tifox — CHANGES_REQUESTED** [review]

File: lib/presentation/widgets/icon_text_button_widget.dart (hunk 1 above)
Comment (KeepCalmAndProgram) [reply lib/presentation/widgets/icon_text_button_widget.dart]: corrected successfully

File: lib/presentation/widgets/text_button_widget.dart (hunk 2 above)
Comment (KeepCalmAndProgram) [reply lib/presentation/widgets/text_button_widget.dart:19]: I do not fully understand how you want to do this and how it should look like in the end.

File: lib/presentation/widgets/text_button_widget.dart (hunk 3 above)
Comment (KeepCalmAndProgram) [reply lib/presentation/widgets/text_button_widget.dart:3]: corrected successfully

Comment (KeepCalmAndProgram) [reply lib/presentation/widgets/text_button_widget.dart]: corrected successfully to stateless widget

File: lib/presentation/widgets/icon_text_button_widget.dart (hunk 4 above)
Comment (KeepCalmAndProgram) [reply lib/presentation/widgets/icon_text_button_widget.dart]: corrected successfully to stateless widget

File: lib/presentation/screens/search_screen.dart (hunk 5 above)
Comment (KeepCalmAndProgram) [reply lib/presentation/screens/search_screen.dart]: corrected successfully

File: lib/presentation/screens/search_screen.dart (hunk 6 above)
Comment (KeepCalmAndProgram) [reply lib/presentation/screens/search_screen.dart]: corrected successfully

File: lib/presentation/screens/search_screen.dart (hunk 7 above)
Comment (KeepCalmAndProgram) [reply lib/presentation/screens/search_screen.dart]: corrected successfully

File: lib/presentation/screens/photo_gallery_screen.dart (hunk 9 above)
Comment (KeepCalmAndProgram) [reply lib/presentation/screens/photo_gallery_screen.dart]: corrected successfully

File: lib/presentation/screens/albums_screen.dart (hunk 10 above)
Comment (KeepCalmAndProgram) [reply lib/presentation/screens/albums_screen.dart]: corrected successfully

File: lib/main.dart (hunk 11 above)
Comment (KeepCalmAndProgram) [reply lib/main.dart:42]: corrected successfully

File: lib/presentation/widgets/text_button_widget.dart (hunk 2 above)
Comment (alex-tifox) [reply lib/presentation/widgets/text_button_widget.dart:19]: Showing you and example on this file how to pass functions as class attributes:

```
class TextButtonCustomise extends StatelessWidget {
  final String text;
  final VoidCallback onPressed;
  
  const TextButtonCustomise({
    Key? key,
    required this.text,
    required this.onPressed,
  }) : super(key: key);

  @override
  Widget build(BuildContext context) {
    return TextButton(
      style: TextButton.styleFrom(
        textStyle: Theme.of(context).textTheme.bodyText1,
      ),
      onPressed: onPressed,
      child: Text(
        text,
        style: Theme.of(context).textTheme.bodyText1,
      ),
    );
  }
}
```

To have better view you can go even to the `TextButton` class or any other widget which have any callback function in it and see, how does it work

**Review by alex-tifox — CHANGES_REQUESTED** [review]

File: lib/presentation/widgets/text_button_widget.dart (hunk 2 above)
Comment (KeepCalmAndProgram) [reply lib/presentation/widgets/text_button_widget.dart:19]: Thanks for the explanation, I agree that creating a callback function is the best practice. Сorrected successfully

**Review by alex-tifox — APPROVED** [review]

**Pull request merged**